documents/development/events.md

```diff
@@ -0,0 +1,108 @@
+# Events
+
+The Grouparoo Event System is modeled to be compatible with a number of existing event & analytic systems.
+
+## Backends
+
+The Grouparoo Event System can use one of the many `EventBackends` that implement the API (TODO). Grouparoo by default ships with `SequelizeEventBackend`, which will reuse the existing connection to your Grouparoo Database in Postgres, Mysql or similar connection.
+
+The backend is responsible for:
+
+- connecting to the data source
+- managing the creation of it’s own tables
+- implementing new classes which extend from `EventPrototype` and `EventBackendPrototype` and implementing the required methods, ie `save` and `static findByGuid`. See `core/api/classes/events` to learn more.
+
+## Schema
+
+The 3 models used for the Grouparoo Event System are: `events`, `eventData` and `apiKeys`. While the specific implementing of the schema used by each backend may differ, a prototypical table structure is modeled below.
+
+### `events` table
+
+This is the primary event table logging a row per event.
+
+- _guid_ - The primary Key for this event. Generated by the Grouparoo server. Should /not/ be auto-incremented but others unique (we use UUIDv4) + a `evt_` prefix.
+- _producerGuid_ - “foreign key” for `apiKeys`. Identifies the API client which created the event, ie: “web API client” vs “iPhone API client”. Set by the grouparoo server.
+- _ipAddress_ - IP address the event was received from. Set by the grouparoo server.
+- _profileGuid_ - “foreign key” for `profiles`. Set by the grouparoo server.
+- _createdAt_ - The time the Grouparoo Server received the event. Set by the grouparoo server.
+- _updatedAt_ - The time the Grouparoo Server last modified the event. Set by the grouparoo server.
+- _occurredAt_ - Set by the client, optional. This likely differs from `createdAt` as it should store the local time of the event. Event’s may be batched when sent from the client, so we cannot rely on the time server received the event to be the time the event occurred. If there is no `occurredAt`, the server will apply `createdAt`
+- _type_ - Set by the client, required. The named type of the event. Examples include `pageview`, `item-added-to-cart`, `purchase`, or `click`
+- _userId_ - Set by the client, optional. If the client knows the primary identifying profile property of the current user, it should be sent. This is /your/ notion of userId. This property takes precedence over `anonymousId` when present.
+- _anonymousId_ - Set by the client, required (unless `userId` is present). If the `userId` is not known (the user is not yet logged in), a unique ID should be generated by the client. These can be device GUIDs in the case of mobile phones, or just another UUID.
+
+### `eventData` table
```

Comment:
should it have `createdAt/updatedAt` ?

Reply:
I think that's up to the backend.  Grouparoo core probably won't do anything with it